util/events/libuv/FileNo.c

```diff
@@ -93,9 +97,17 @@ static void incoming(uv_pipe_t* stream,
     } else {
         struct FileNo_pvt* fileno = Identity_check((struct FileNo_pvt*) stream->data);
         if (fileno->peer.accepted_fd != -1) {
+            int oldFd = fileno->peer.accepted_fd;
+            int newFd = 123;
```

Comment:
So the `dup()` is actually necessary. When we were working on this, `oldFd` kept getting closed and we couldn't figure out who is closing it. As a last-ditch effort, @lgierth cloned it before it vanishes and `newFd` the one that gets used.

See: https://github.com/cjdelisle/cjdns/pull/971/files#diff-87204fd2d5d9c342a032ce51522fd6cdR109